### PR TITLE
3672 request to join

### PR DESCRIPTION
**Pull request by cuahsimarko**
- Fixes https://github.com/hydroshare/hydroshare/issues/3636
- Fixes https://github.com/hydroshare/hydroshare/issues/3672

This has been deployed to http://dev-hs-7.cuahsi.org/ and due to the complexity of manual test cases, CUAHSI will coordinate the testing effort outside of this PR.

### CODE REVIEW

File: hs_communities/views/communities.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def get_context_data(self, **kwargs):
         grpfilter = self.request.GET.get('grp')
 
         community = community_from_name_or_id(kwargs['community_id'])
-        community_resources = community.public_resources
+        community_resources = community.public_resources.distinct()
         raw_groups = community.groups_with_public_resources()
```
Comment (cuahsimarko) [inline hs_communities/views/communities.py]: @alvacouch per our discussion about beta

File: theme/templates/pages/group.html [hunk 2]
```diff
@@ -381,36 +391,37 @@ <h4><strong>{{ pendingrequest.invitation_to|contact }}</strong>
                     </div>
                 </div>
 
-                {% if profile_user.is_group_owner or profile_user.is_group_editor or profile_user.is_group_viewer %}
-                    <div role="tabpanel" class="tab-pane fade in active" id="resources">
+                <div role="tabpanel" class="tab-pane fade in active" id="resources">
```
Comment (cuahsimarko) [inline theme/templates/pages/group.html]: Show the Resources tab regardless of permissions held by the viewer; resource listing will be restricted further down. Overview tab has been removed.

File: theme/templates/pages/group.html [hunk 3]
```diff
@@ -381,36 +391,37 @@ <h4><strong>{{ pendingrequest.invitation_to|contact }}</strong>
...
-                                            {% for res in group_resources %}
-                                                <span data-grantor-id="{{ res.grantor.id }}"
-                                                      data-grantor-name="{{ res.grantor|best_name }}"></span>
-                                            {% endfor %}
-                                        </div>
-                                    </div>
-                                </div>
-                            </div>
-                        </div>
+                         {% if profile_user.is_group_owner or profile_user.is_group_editor or profile_user.is_group_viewer %}
```
Comment (cuahsimarko) [inline theme/templates/pages/group.html]: Don't show resources if not a Public Group and the user viewing the page does not have access

File: theme/templates/pages/group.html [hunk 4]
```diff
@@ -569,10 +580,57 @@ <h4 class="panel-title">
 
                             {% include "includes/legend.html" %}
 
-                        </div>
-                     </div>
-                 </div>
+                         {% else %}
```
Comment (cuahsimarko) [inline theme/templates/pages/group.html]: If not a member of the Group provide option to join or display join request status

**Comment by dtarb** [comment]
@cuahsimarko How do I get an account on http://dev-hs-7.cuahsi.org/ to test.  I just requested an account but did not receive confirmation email either at my email address or the squirrelmail testing site.

**Comment by Maurier** [comment]
@dtarb I have now activated your account through the admin page.

File: theme/templates/pages/group.html [hunk 5]
```diff
@@ -569,10 +580,57 @@ <h4 class="panel-title">
...
+                                    {% csrf_token %}
+                                    <button type="submit" class="btn btn-primary" role="button">Ask to join</button>
+                                </form>
+                            {% endif %}
+                                {% else %}
+                                <h4><i>This group is currently inactive.</i></h4>
+                            {% endif %}
+</div>
                 {% endif %}
+
```
Comment (Maurier) [inline theme/templates/pages/group.html]: There are some indentation issues here.

**Comment by cuahsimarko** [comment]
I had created a few new accounts for David this morning and emailed him the information instead of posting passwords here. I've done the same for Neal.

File: theme/templates/pages/group.html [hunk 6]
```diff
@@ -89,23 +89,33 @@ <h4 class="text-muted">{{ group.gaccess.purpose|linebreaks }}</h4>
         <div class="full-width-tabs-container" id="user-profile-tabs">
             <!-- Nav tabs -->
             <ul class="nav nav-tabs full-width-tabs" role="tablist">
-                {% if profile_user.is_group_owner or profile_user.is_group_editor or profile_user.is_group_viewer %}
+{#                {% if profile_user.is_group_owner or profile_user.is_group_editor or profile_user.is_group_viewer %}#}
```
Comment (Maurier) [inline theme/templates/pages/group.html]: Remove this commented out code and empty lines below if not needed anymore.

**Comment by cuahsimarko** [comment]
> Looks good. Some code clean up is needed to fix indentations and remove extra empty lines.

I had done the syntax cleanup and it made the PR unreadable. Now that people have looked at it, I'll push a syntax only commit.

File: theme/templates/pages/group.html (hunk 5 above)
Comment (cuahsimarko) [reply theme/templates/pages/group.html]: Fixed in 4c23d114c

File: theme/templates/pages/group.html (hunk 6 above)
Comment (cuahsimarko) [reply theme/templates/pages/group.html]: Fixed in 4c23d114c

**Comment by cuahsimarko** [comment]
@alvacouch @Maurier can you please approve as soon as you get a chance? This is important to get out in the imminent beta release, which could be as early as Monday Nov 18

**Pull request merged**